src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -585,4 +616,82 @@ private String mapToType(Reason reason) throws CTPException {
         throw new CTPException(Fault.SYSTEM_ERROR, "Unexpected refusal reason: %s", reason);
     }
   }
+
+  /**
+   * Make Case Service request to return cases by UPRN
+   *
+   * @param uprn of requested cases
+   * @param listCaseEvents boolean of whether require case events
+   * @return List of cases for UPRN
+   * @throws CTPException
+   */
+  private List<CaseDTO> callCaseSvcByUPRN(Long uprn, Boolean listCaseEvents) throws CTPException {
+
+    List<CaseContainerDTO> rmCases = new ArrayList<>();
+    try {
+      rmCases = caseServiceClient.getCaseByUprn(uprn, listCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with(uprn).info("Case by UPRN Not Found calling Case Service");
+        return Collections.emptyList();
+      } else {
+        log.with(uprn).with("status", ex.getStatus()).error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    // Only return cases that are not of caseType = HI
+    List<CaseContainerDTO> casesToReturn =
+        (List<CaseContainerDTO>)
+            rmCases
+                .parallelStream()
+                .filter(c -> !(c.getCaseType().equals(CaseType.HI.name())))
```

Comment:
Case type should never be null, but if it should ever happen to be null then it might be better to fail with an error than to silently keep going.

Reply:
> Again , sorry to annoy you about code you have moved in this PR rather than originated, but the enum name will never be null, but the caseType() call conceivably could be, so safer code would reverse the the equals, ie CaseType.HI.name().equals(c.getCaseType()).

Done, minor change, guards against NullpointerException if CaseType null.

Reply:
> Case type should never be null, but if it should ever happen to be null then it might be better to fail with an error than to silently keep going.

Can't throw a Checked exception from within the filter Predicate condition so would have to throw some form of RunTimeException. Either surround this then in a "try{ } catch (){}" block and do something like throw a CTPException then or let the RunTimeException go back to the client. Is this a scenario that will happen? Please give more details if think need to error on CaseType being null.

Reply:
I wasn't suggesting throwing an exception, but actually arguing for the code to be left unchanged (I should have been clearer about that).
Switching the sides for the equals does of course prevent a NullPointer Exception but is that really what we want? If caseType were to ever be null then we do still have a problem but we just wouldn't know about it.

Reply:
Apologies, maybe you and Rob could discuss. Guess the question is if CaseType is null is it legitimate for it to be returned to the client, or totally spurious, we don't know what it is and should return a NullPointerExcpetion, which we should probably catch and turn into a CTPException with Fault.SYSTEM_ERROR. That is the original predicate condition wrapped in a try{}catch(){} that throws the CTPException?

Reply:
I personally preferred the code as it was. But surely the potential NPE doesn't need any code to support it (after all it really should never happen!). 
Any part of the CC codebase can throw a NPE exception so the existing CC exception handling ought to be able to do the right thing with the NPE (and if not then that should be the responsibility of a separate CR).
Now having said that I don't feel super strongly about. It's obviously not a huge issue so will go with whatever. I just wanted to vote in favour of the original code, and point out the downside of never failing is that when the impossible happens you don't know that something has gone wrong.

Reply:
I'm not going to sweat this one two much. If getCaseType() can never be null it doesn't make so much difference, although as a general rule I would just naturally try to write null tolerant code to protect against those expressions where one side can be null.

I'll leave it to Martin to decide!

Reply:
Left as is by agreement.